### PR TITLE
Closes #1627 1627_exprs@208_exprs@devel: replace `vars()` with `exprs()`

### LINKED ISSUE #1627: General Issue: Replace `vars()`
### Background Information

The `vars()` function is used extensively in admiral and admiral function calls. Unfortunately, it is superseded (see https://dplyr.tidyverse.org/reference/index.html#superseded).

### Definition of Done

- [x] Team agrees on a replacement of `vars()`.
- [x] Develop deprecation strategy and implement
- [x] Replacement in functions
- [x] Investigate templates and replace where necessary
- [x] Investigate vignettes and replace where necessary
- [x] Investigate tests and replace where necessary

@bundfussr Can you think of anywhere else that we need to replace `vars()` or be worried about?

### PR DESCRIPTION
**Pull request by bundfussr**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bundfussr** [comment]
@galachad , I have implemented a hook for forcing the admiral definition of `vars()`. Could you have a look?

**Comment by bundfussr** [comment]
@galachad , @thomas-neitmann , @bms63, unfortunately, placing `vars()` into the global environment causes the R-CMD check to fail. It is also a violation of the "CRAN Repository Policy".

Any ideas?

Instead of defining `vars()` in the global environment we could just issue a message like 
>dplyr was attached after admiral. If you  are using `vars()` in your admiral function calls, attach admiral after dplyr or replace the `vars()` calls by `exprs()` calls.

What do you think?

**Comment by bundfussr** [comment]
@galachad , @thomas-neitmann , @bms63 , @cicdguy , the man page of `reexports.R` is causing an issue. It contains the line
```
\item{rlang}{\code{\link[rlang:defusing-advanced]{exprs}}}
```
which causes the R-CMD 4.0 check to fail.
I can fix it by replacing the line with
```
\item{rlang}{\code{\link[rlang]{exprs}}}
```
but then the Man Pages check fails.

Any ideas?

**Comment by zdz2101** [comment]
Is the best course of action to push both `admiral` and `admiraldev` changes at the same time when we are ready? And am I right to think during the review process I should run: 

remotes::install_github("pharmaverse/admiraldev", ref = "208_exprs@devel") 
 
whilst checking this branch's changes?

**Comment by bundfussr** [comment]
@zdz2101 , yes to both of your questions.

File: R/reexports.R [hunk 1]
```diff
@@ -5,7 +5,59 @@
...
+#' @export
+vars <- function(...) {
+  deprecate_warn(
+    "0.10.0",
+    "vars()",
+    "exprs()",
+    details = paste(
+      "The admiral functions no longer expect list of quosures created by `vars()`",
+      "but list of expressions created by `exprs()`"
+    )
```
Comment (bms63) [inline R/reexports.R:33]:
```suggestion
      "The admiral functions no longer expects list of quosures created by `vars()`",
      "but list of expressions created by `exprs()`"
    )
```

Comment (bms63) [reply R/reexports.R:33]: Can we advise admiral users to update in the message as well- with like a please update ASAP message or something concise??

**Comment by bms63** [comment]
> @galachad , @thomas-neitmann , @bms63 , @cicdguy , the man page of `reexports.R` is causing an issue. It contains the line
> 
> ```
> \item{rlang}{\code{\link[rlang:defusing-advanced]{exprs}}}
> ```
> 
> which causes the R-CMD 4.0 check to fail. I can fix it by replacing the line with
> 
> ```
> \item{rlang}{\code{\link[rlang]{exprs}}}
> ```
> 
> but then the Man Pages check fails.
> 
> Any ideas?

I feel like this has happened before in our repo but I can not find it in any PRs or issues.  @pharmaverse/admiral any memories??  It is strange that the `\item{rlang}{\code{\link[rlang]{exprs}}}` fails on our `man` action as that action uses 4.0.  I googled around and do not see anyone else struggling with this issue.

Also, is it worrisome that the `(remotes::install_github("pharmaverse/admiraldev", ref = "208_exprs@devel"))` update I did locally doesn't seem to be impacting our Actions.  Did you do something behind the scenes?

**Comment by bundfussr** [comment]
@bms63 , yes, we had this issue before. I think it was before we introduced the Man check and when we were using R 3.6. At this point in time we could work-around the issue by editing the generated man files.

Regarding the admiraldev version, the checks are using the `208_exprs` branch of admiraldev because of the "@208_exprs" part in the branch name. This is a feature of staged dependencies.

File: R/reexports.R (hunk 1 above)
Comment (bundfussr) [reply R/reexports.R]: Yes, I have extended the message.

**Comment by bms63** [comment]
My mind is blown that you can do this - I didn't even see this in the branch name.  Way cool.

I do something with the github actions and the branch names, but this opens doors for me!!  

image

**Comment by zdz2101** [comment]
Also, on that reexports topic, the repercussions of using noRd for that specific entry would be that it simply isn't logged in our "Objects exported from other packages" page right? Which is only accessible through the Help window on an individual's R Session, not our pkgdown website?

File: tests/testthat/test-admiral_options.R [hunk 2]
```diff
@@ -18,13 +18,13 @@ test_that("get_admiral_option Test 3: non-character argument triggers assertion
...
 test_that("set_admiral_options Test 4: set works", {
-  set_admiral_options(subject_keys = vars(STUDYID, USUBJID2))
-  expect_equal(get_admiral_option("subject_keys"), vars(STUDYID, USUBJID2))
+  set_admiral_options(subject_keys = exprs(STUDYID, USUBJID2))
+  expect_equal(get_admiral_option("subject_keys"), exprs(STUDYID, USUBJID2))
 })
 
 ## Test 5: unexpected function input for set gives error ----
 test_that("set_admiral_options Test 5: unexpected function input for set gives error", {
   expect_error(set_admiral_options(subject_keys = quo_c(STUDYID, USUBJID2)))
```
[the remark this answers is not in the document]
Comment (bundfussr) [reply tests/testthat/test-admiral_options.R]: Using `expr_c()` causes the same error as in the next line. Therefore I have updated to `quos()` and added the expected error messages.

**Comment by bundfussr** [comment]
Using `@noRd` would avoid the issues in the checks. However, we would hide that we reexport `exprs()`. I think we shouldn't do this. I also wonder if we should add a "Reexported" section to the reference page.

**Comment by zdz2101** [comment]
Mentioning https://github.com/pharmaverse/admiral/issues/1706 for the decision to put a reexports section in our reference page

**Comment by bundfussr** [comment]
@bms63 , @thomas-neitmann , @zdz2101 , how should we proceed with this PR? Is there anything outstanding except the failing check?

The check is only failing in the github workflow. If I run it locally, it passes. Furthermore, the link which causes the check to fail, does not cause any issues. It is working in the RStudio help tab and in the HTML version.

We have a separate issue for the failing check (https://github.com/pharmaverse/admiral/issues/1706). Should we ignore it for this PR and merge it?

**Comment by zdz2101** [comment]
The earlier we let this go through the more room for error/time we have before release too, so I'm inclined to say go for it

**Review by zdz2101 — APPROVED** [review]

**Review by bms63 — APPROVED** [review]

**Comment by galachad** [comment]
The problem with 4.0 check is caused by `rlang`. The cmd check for 4.0 is done with `rlang` 0.4.x. In `rlang` 1.0.x the `nse-defusing` was tweak to `defusing` (https://github.com/r-lib/rlang/commit/65e6d3072c1b341102809d932f2dee91a118e8ab). 

The warning error returns by the check are correct as `defusing-advanced` do not exist in `rlang` 0.4.x. This is probably the first compatibility error that occurs.

It can be fixed by upgrade the `rlang` or modifying the r cmd check to re-render documentation, but at the end the links for the user using `rlang` 0.4.x will be broken. This documentation issue is not a big deal, but we should decide when to migrate to rlang `1.0` if possible.

**Comment by galachad** [comment]
Currently, the best solution will be to export the `exprs` as @zdz2101 suggest on #1706. 

The `\link[rlang:exprs]{exprs}` will not be replaced by section, so should be compatible with rlang 0.4

Example:
```
#' rlang exprs
#'
#' See \code{rlang::\link[rlang:exprs]{exprs}} for details.
#'
#' @name exprs
#' @rdname reexport-exprs
#' @keywords reexport
#' @importFrom rlang exprs
#' @export
NULL
```

**Comment by bundfussr** [comment]
Thanks @galachad and @zdz2101 ! I have updated reexports.R as suggested.

Unfortunately, all workflows are failing at the moment. Seems to be a github issue. I'll try to rerun them later.

**Comment by bundfussr** [comment]
@bms63 , @zdz2101 , could you reapprove to start the check templates check?

**Review by bms63 — APPROVED** [review]

**Pull request merged**